### ISSUE TITLE
Support "Deploy from Image" flow

### ISSUE THREAD

**Issue by ALRubinger** [fabric8-launcher/launcher-planning#199]
k8s and OpenShift are orchestration platforms for containers.

A user should be able to simply say "here's my image that I or my existing toolchain built.  Run it".

Do with this with straight `oc` now you need a template and other knowledge.  Make this a 0-knowledge-required operation.  If you have an image from Docker or Buildah or whatever, deploy the thing.

**Comment by quintesse** [comment]
The OpenShift console itself already has this functionality, right? Just confirming that we want the same in the Launcher.

**Comment by ALRubinger** [comment]
Does it?  AriL told me yesterday it didn't, but maybe we missed something?

>

**Comment by ALRubinger** [comment]
![screenshot37](https://user-images.githubusercontent.com/199891/51391749-ebb57b00-1b00-11e9-9f44-eee86c98a54f.png)

No upload.  Assumes a tag or registry item is already in place.

**Comment by quintesse** [comment]
Ok, but images always come from a registry, right? AFAIK there's no (standardized) image file that you can create from a docker/buildah image, or is there?

**Comment by gastaldi** [comment]
Yes, images always come from a registry. There isn't a file representing the image afaik.

**Comment by ALRubinger** [comment]
So why not from the output of: https://docs.docker.com/engine/reference/commandline/save/

**Comment by ALRubinger** [comment]
Maybe not.  But maybe it would if it bridges the gap between "hey I have
this image I made through my my own means" and "OpenShift go deploy it".

It's a workflow which grants full control to the developer over
containerization and build and lets them use OpenShift to deploy the result.

>

**Comment by ppitonak** [comment]
@ALRubinger IMHO there are two separate groups of applications

* single-container apps (optionally, using externally managed services) - you just build the app image anyhow and import it into the OpenShift, you need to create secrets in OpenShift in the case of private docker registry (annoying but relatively easy, one-time thing)
* multi-container apps - you need to describe your application somehow

IMHO the deployment of single-container apps is a solved problem. Is this issue about multi-container apps?
